bigtop-packages/src/common/hive/do-component-build

```diff
@@ -32,11 +32,21 @@ HIVE_MAVEN_OPTS=" -Dhbase.version=$HBASE_VERSION \
 # Include common Maven Deployment logic
 . $(dirname ${0})/maven_deploy.sh
 
+# workaround for BIGTOP-3548, as the protoc binary included in
+# com.github.os72::protoc doesn't work with legacy centos-7 libc
+# so we use protoc built by bigtop_toolchain to create a fake jar
+source /etc/os-release
+case ${ID}-${VERSION_ID} in
+    centos-7)
+        mvn install:install-file -DgroupId=com.github.os72 -DartifactId=protoc -Dversion=2.6.1-build3 \
```

Comment:
protoc built by bigtop_toolchain is 2.5.0. If we can use local protoc artifact, how about set the version back to 2.5.0 for all distros/archs?

Reply:
That should work. Actually, I'm a bit hesitating on which way we need to use: just create a local 2.5.0 artifact or backport upstream patch (HIVE-21939) while making a hack for CentOS-7 as well. I'm just wondering if there is a possibility that a user directly builds the source and patches in srpm/sdeb without bigtop toolchain. If that is a case it might be better to align with HIVE upstream as much as possible while creating a little hack here in the build script.

Reply:
Hadoop already depend on protoc built by bigtop_toolchain (on aarch64 and ppc64le). Phoenix will do so by #786. Asking users for using bigtop_toolchain on aarch64 and ppc64le should be reasonable. Since the fix version of HIVE-21939 is 4.0.0, it is not tested on the field. Removing patch6-HIVE-21939.diff should be safer if possible.

Reply:
Ya. It makes sense to use the same solution to fix the same protoc problem across hive/hbase/phoenix. Will update patch soon.